packages/web-pkg/src/components/AppBar/AppBar.vue

```diff
@@ -0,0 +1,341 @@
+<template>
+  <div id="files-app-bar" ref="filesAppBar" :class="{ 'files-app-bar-squashed': sideBarOpen }">
+    <oc-hidden-announcer :announcement="selectedResourcesAnnouncement" level="polite" />
+    <div class="files-topbar oc-py-s">
+      <h1 class="oc-invisible-sr" v-text="pageTitle" />
+      <div
+        class="oc-flex oc-flex-middle files-app-bar-controls"
+        :class="{
+          'oc-flex-between': breadcrumbs.length || hasSharesNavigation,
+          'oc-flex-right': !breadcrumbs.length && !hasSharesNavigation
+        }"
+      >
+        <oc-breadcrumb
+          v-if="showBreadcrumb"
+          id="files-breadcrumb"
+          data-testid="files-breadcrumbs"
+          class="oc-flex oc-flex-middle"
+          context-menu-padding="small"
+          :show-context-actions="showContextActions"
+          :items="breadcrumbs"
+          :max-width="breadcrumbMaxWidth"
+          :truncation-offset="breadcrumbTruncationOffset"
+          @item-dropped-breadcrumb="fileDroppedBreadcrumb"
+        >
+          <template #contextMenu>
+            <context-actions
+              :action-options="{ space, resources: breadcrumbsContextActionsItems.filter(Boolean) }"
+            />
+          </template>
+        </oc-breadcrumb>
+        <portal-target v-if="showMobileNav" name="app.runtime.mobile.nav" />
+        <slot v-if="hasSharesNavigation" name="navigation" />
```

Comment:
I think we can remove `hasSharesNavigation` as property and replace it with a `computed` that checks, if the slot `navigation` is given.